tools/ipapy2/ipapy2_map_isotope_patterns.xml

```diff
@@ -0,0 +1,62 @@
+<tool id="ipapy2_map_isotope_patterns" name="IPA map isotope patterns" version="@TOOL_VERSION@+galaxy0" profile="21.05">
+    <macros>
+        <import>macros.xml</import>
+    </macros>
+    
+    <requirements>
+        <requirement type="package" version="@TOOL_VERSION@">ipapy2</requirement>
+    </requirements>
+
+    <command detect_errors="exit_code"><![CDATA[
+        python3 '${ipapy2_map_isotope_cli}'
+    ]]></command>
+
+<configfiles>
+<configfile name="ipapy2_map_isotope_cli">
+@init_import@
+
+clustered_df = pd.read_csv('${clustered_MS1_intensities}')
+<!-- The function call below updates the passed dataframe in place hence it returns nothing -->
+ipa.map_isotope_patterns(clustered_df, isoDiff=$isotope.isoDiff, ppm=$isotope.ppm, ionisation=int(${ionisation}), MinIsoRatio=$isotope.isotope_ratio)
+clustered_df.to_csv('${mapped_isotope_patterns}', index=False)
+</configfile>
+</configfiles>
+
+    <inputs>
+        <param label="IPA dataframe" name="clustered_MS1_intensities" type="data" format="csv" help="A csv file of clustered MS1 intensities." />
+        <expand macro="ionisation"/>
+        <section name="isotope" title="isotope settings">
+            <param name="isoDiff" type="float" value="1">
+                <label>isotope difference</label>
+                <help>Difference between isotopes of charge 1.</help>
+            </param>
+            <param name="ppm" type="float" value="100">
+                <label>ppm between isotopes</label>
+                <help>Maximum ppm value allowed between 2 isotopes. Default value 100.</help>
+            </param>
+            <param name="isotope_ratio" type="float" min="0" max="100" value="0.5">
+                <label>isotope ratio</label>
+                <help>mininum intensity ratio expressed (Default value 0.5%). 
+				Only isotopes with intensity higher than MinIsoRatio% of the main isotope are considered</help>
+            </param>
+        </section>
+    </inputs>
+    
+    <outputs>
+         <data label="${tool.name} on ${on_string}" name="mapped_isotope_patterns" format="csv"/>
+   </outputs>
+    
+    <tests>
+        <test>
+            <param name="clustered_MS1_intensities" value="clustered_MS1_intensities.csv"/>
+            <output name="mapped_isotope_patterns" file="mapped_isotope_patterns.csv"/>
+        </test>
+    </tests>
+
+    <help><![CDATA[
+	::
+		mapping isotope patterns in MS1 data.
```

Comment:
This can probably be expanded - I think also large parts of the help texts can be shared across tools in a separate macro file and used as tokens to be instantiated where needed, for example descriptions of the input tables etc., which column names are expected and so on and so forth.